File: ios/PosPrinter.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(PosPrinter, NSObject)

RCT_EXTERN_METHOD(checkStatus:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(open:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(setGrey:(int)level)

RCT_EXTERN_METHOD(setLineSpace:(int)lineSpace)

RCT_EXTERN_METHOD(setBold:(BOOL)isBold)

RCT_EXTERN_METHOD(setAlgin:(int)mode)

RCT_EXTERN_METHOD(setTextSize:(int)size)

RCT_EXTERN_METHOD(addString:(NSString *)content)

RCT_EXTERN_METHOD(printString)

RCT_EXTERN_METHOD(walkPaper:(int)line)

RCT_EXTERN_METHOD(printLogo:(NSString *)image withIsBuffer:(BOOL)isBuffer)

RCT_EXTERN_METHOD(printHTML:(NSString *)html
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(close:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/PosPrinter.swift
import Foundation

@objc(PosPrinter)
class PosPrinter: NSObject {
    
  @objc(checkStatus:withRejecter:)
  func checkStatus(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
    resolve(0)
  }
    
  @objc(open:withRejecter:)
  func open(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
    resolve(true)
  }
  
  @objc(setGrey:)
  func setGrey(level: Int) -> Void {
    
  }
  
  @objc(setLineSpace:)
  func setLineSpace(lineSpace: Int) -> Void {
    
  }
  
  @objc(setBold:)
  func setBold(isBold: Bool) -> Void {
    
  }
  
  @objc(setAlgin:)
  func setAlgin(mode: Int) -> Void {
    
  }
  
  @objc(setTextSize:)
  func setTextSize(size: Int) -> Void {
    
  }
  
  @objc(addString:)
  func addString(content: String) -> Void {
    
  }
  
  @objc(printString)
  func printString() -> Void {
    
  }
  
  @objc(walkPaper:)
  func walkPaper(line: Int) -> Void {
    
  }
  
  @objc(printLogo:withIsBuffer:)
  func printLogo(image: String, isBuffer: Bool = false) -> Void {
    
  }

  @objc(printHTML:withResolver:withRejecter:)
  func printHTML(html: String, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
    resolve(true)
  }

  @objc(close:withRejecter:)
  func close(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
    resolve(true)
  }
}
